main.tf

```diff
@@ -21,10 +21,10 @@ resource "azurerm_network_security_rule" "predefined_rules" {
   direction                   = "${element(var.rules["${lookup(var.predefined_rules[count.index], "name")}"], 0)}"
   access                      = "${element(var.rules["${lookup(var.predefined_rules[count.index], "name")}"], 1)}"
   protocol                    = "${element(var.rules["${lookup(var.predefined_rules[count.index], "name")}"], 2)}"
-  source_port_ranges          = "${split(",", replace(  "${lookup(var.predefined_rules[count.index], "source_port_range", "*" )}"  ,  "*" , "0-65535" ) )}"
+  source_port_range           = "${lookup(var.predefined_rules[count.index], "source_port_range", "*" )}"
```

Comment:
Are we sure that we will never need multiple ranges?

Reply:
if you need this much configurability, use the `custom_rule` attribute. I think we should keep the `predefined_rules` as simple as possible